src/object.d

```diff
@@ -3287,135 +3302,57 @@ int __cmp(T1, T2)(T1[] s1, T2[] s2)
         }
         return s1.length < s2.length ? -1 : (s1.length > s2.length);
     }
-    // floating point types
-    else static if (__traits(isFloating, U1))
+    // Imaginary types
+    else static if (is(U1 == ifloat) || is(U1 == idouble) || is(U1 == ireal))
     {
+        // Special-case imaginary types to use comparison for "concrete" types
         static if (is(U1 == ifloat)) alias F = float;
         else static if (is(U1 == idouble)) alias F = double;
         else static if (is(U1 == ireal)) alias F = real;
```

Comment:
Oops, that was a reply to a different comment (normalizing constness). In which case the action item is either `static assert(is(T1 == const(U1)) && is(T2 == const(U2)))` or `static if (!is(T1 == const(U1)) || !is(T2 == const(U2)) { return __cmp(cast(const)s1, cast(const)s2)); } else { ...  }` OSLT.